src/collaborate.rs

```diff
@@ -129,19 +169,74 @@ impl Collaboration {
         msg_type: CommsMessageType,
         data: Value,
         program_context: &ProgramContext,
+        timestamp: i64,
+        signature: Vec<u8>,
     ) -> Result<(), BitVMXError> {
         let pubkey_hash = comms_address.pubkey_hash.clone();
         match msg_type {
+            CommsMessageType::VerificationKey => {
+                // Handle peer verification key message
+                let verification_key: String = serde_json::from_value(data.clone())
+                    .map_err(|_| BitVMXError::InvalidMessageFormat)?;
+
+                // Reconstruct the message that was signed: {program_id}{msg_string}{timestamp}
+                // Serialize with sorted keys to match the signing logic
+                let message = crate::comms_helper::construct_message(
+                    &self.collaboration_id.to_string(),
+                    &data,
+                    timestamp,
+                )?;
+
+                let verified = program_context.key_chain.verify_rsa_signature(
+                    &verification_key,
+                    message.as_bytes(),
+                    &signature,
+                )?;
+                info!(
+                    "Verification key verified for peer: {} ({})",
+                    pubkey_hash, verified
+                );
+                self.participant_verification_keys
```

Comment:
We are verifying the verification key message with the RSA key contained in the same message , it is kind of self-attestation and a MITM can swap the whole message with its own key and get it accepted. We should sign the VerificationKey message (e.g. {program_id || rsa_pubkey || pubkey_hash || timestamp})  with a different key, maybe the communications key. Also reject when verify_rsa_signature returns false. We should also reject the key when verify_rsa_signature returns false, right now we are just logging the result of the verification.